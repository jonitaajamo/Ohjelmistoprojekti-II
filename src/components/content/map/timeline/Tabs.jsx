import React, { Component } from "react";
import Button from "./tabs/Button";

export default class Tabs extends Component {
  constructor() {
    super();
    this.state = {
      active: "all"
    };
  }

  clickHandler(name, key) {
    this.setState({
      active: name
    });
    this.props.onAssetChange(key);
  }

  render() {
    const buttons = this.props.geographicalWeightData.length
      ? this.props.geographicalWeightData[0].assetClasses.map((item, key) => {
          return (
            <Button
              isActive={item.class === this.state.active ? "is-active" : ""}
              onClickHandler={() => this.clickHandler(item.class, key)}
              key={key}
              asset={item.class}
            />
          );
        })
      : [];

    return (
      <div className="tabs is-toggle is-toggle-rounded is-centered is-fullwidth ">
        {buttons}
      </div>
    );
  }
}
