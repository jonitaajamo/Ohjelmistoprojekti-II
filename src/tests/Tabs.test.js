import React from "react";
import Tabs from "../components/content/map/timeline/Tabs";

describe("Tabs", () => {
  const geographicalWeightData = [
    {
      month: "2012-01",
      assetClasses: [
        {
          class: "all",
          weights: [
            { countryId: 840, weight: 0.7, marketValue: 14000000 },
            { countryId: 246, weight: 0.2, marketValue: 4000000 },
            { countryId: 826, weight: 0.05, marketValue: 1000000 },
            { countryId: 643, weight: 0.05, marketValue: 1000000 }
          ]
        },
        {
          class: "equity",
          weights: [
            { countryId: 276, weight: 0.1, marketValue: 3000000 },
            { countryId: 246, weight: 0.9, marketValue: 27000000 }
          ]
        },
        {
          class: "bond",
          weights: [
            { countryId: 276, weight: 0.1, marketValue: 2000000 },
            { countryId: 246, weight: 0.9, marketValue: 18000000 }
          ]
        },
        {
          class: "alternative",
          weights: [
            { countryId: 276, weight: 0.1, marketValue: 5000000 },
            { countryId: 246, weight: 0.2, marketValue: 10000000 },
            { countryId: 826, weight: 0.3, marketValue: 15000000 },
            { countryId: 643, weight: 0.4, marketValue: 20000000 }
          ]
        },
        {
          class: "property",
          weights: [
            { countryId: 276, weight: 0.1, marketValue: 6000000 },
            { countryId: 246, weight: 0.2, marketValue: 12000000 },
            { countryId: 826, weight: 0.3, marketValue: 18000000 },
            { countryId: 643, weight: 0.4, marketValue: 24000000 }
          ]
        },
        {
          class: "cash",
          weights: [
            { countryId: 276, weight: 0.1, marketValue: 6000000 },
            { countryId: 246, weight: 0.2, marketValue: 12000000 },
            { countryId: 826, weight: 0.3, marketValue: 18000000 },
            { countryId: 643, weight: 0.4, marketValue: 24000000 }
          ]
        },
        {
          class: "unknown",
          weights: [
            { countryId: 276, weight: 0.1, marketValue: 6000000 },
            { countryId: 246, weight: 0.2, marketValue: 12000000 },
            { countryId: 826, weight: 0.3, marketValue: 18000000 },
            { countryId: 643, weight: 0.4, marketValue: 24000000 }
          ]
        }
      ]
    }
  ];

  it("renders correctly", () => {
    const tabs = shallow(
      <Tabs geographicalWeightData={geographicalWeightData} />
    );
    expect(tabs).toMatchSnapshot();
    const buttons = tabs.find("Button");
    expect(buttons.length).toBe(7);
  });

  it("sets asset names correctly", () => {
    const tabs = shallow(
      <Tabs geographicalWeightData={geographicalWeightData} />
    );
    const buttons = tabs.find("Button");
    const texts = buttons.map(node =>
      node
        .dive()
        .find("span")
        .text()
    );
    expect(texts).toEqual([
      "All",
      "Equity",
      "Bond",
      "Alternative",
      "Property",
      "Cash",
      "Unknown"
    ]);
  });

  it("initially sets 'All' as active tab", () => {
    const tabs = shallow(
      <Tabs geographicalWeightData={geographicalWeightData} />
    );
    expect(
      tabs
        .find("Button")
        .at(0)
        .props().isActive
    ).toEqual("is-active");
  });
});
