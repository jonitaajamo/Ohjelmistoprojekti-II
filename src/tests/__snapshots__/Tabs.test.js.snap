// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tabs
    geographicalWeightData={
        Array [
            Object {
              "assetClasses": Array [
                Object {
                  "class": "all",
                  "weights": Array [
                    Object {
                      "countryId": 840,
                      "marketValue": 14000000,
                      "weight": 0.7,
                    },
                    Object {
                      "countryId": 246,
                      "marketValue": 4000000,
                      "weight": 0.2,
                    },
                    Object {
                      "countryId": 826,
                      "marketValue": 1000000,
                      "weight": 0.05,
                    },
                    Object {
                      "countryId": 643,
                      "marketValue": 1000000,
                      "weight": 0.05,
                    },
                  ],
                },
                Object {
                  "class": "equity",
                  "weights": Array [
                    Object {
                      "countryId": 276,
                      "marketValue": 3000000,
                      "weight": 0.1,
                    },
                    Object {
                      "countryId": 246,
                      "marketValue": 27000000,
                      "weight": 0.9,
                    },
                  ],
                },
                Object {
                  "class": "bond",
                  "weights": Array [
                    Object {
                      "countryId": 276,
                      "marketValue": 2000000,
                      "weight": 0.1,
                    },
                    Object {
                      "countryId": 246,
                      "marketValue": 18000000,
                      "weight": 0.9,
                    },
                  ],
                },
                Object {
                  "class": "alternative",
                  "weights": Array [
                    Object {
                      "countryId": 276,
                      "marketValue": 5000000,
                      "weight": 0.1,
                    },
                    Object {
                      "countryId": 246,
                      "marketValue": 10000000,
                      "weight": 0.2,
                    },
                    Object {
                      "countryId": 826,
                      "marketValue": 15000000,
                      "weight": 0.3,
                    },
                    Object {
                      "countryId": 643,
                      "marketValue": 20000000,
                      "weight": 0.4,
                    },
                  ],
                },
                Object {
                  "class": "property",
                  "weights": Array [
                    Object {
                      "countryId": 276,
                      "marketValue": 6000000,
                      "weight": 0.1,
                    },
                    Object {
                      "countryId": 246,
                      "marketValue": 12000000,
                      "weight": 0.2,
                    },
                    Object {
                      "countryId": 826,
                      "marketValue": 18000000,
                      "weight": 0.3,
                    },
                    Object {
                      "countryId": 643,
                      "marketValue": 24000000,
                      "weight": 0.4,
                    },
                  ],
                },
                Object {
                  "class": "cash",
                  "weights": Array [
                    Object {
                      "countryId": 276,
                      "marketValue": 6000000,
                      "weight": 0.1,
                    },
                    Object {
                      "countryId": 246,
                      "marketValue": 12000000,
                      "weight": 0.2,
                    },
                    Object {
                      "countryId": 826,
                      "marketValue": 18000000,
                      "weight": 0.3,
                    },
                    Object {
                      "countryId": 643,
                      "marketValue": 24000000,
                      "weight": 0.4,
                    },
                  ],
                },
                Object {
                  "class": "unknown",
                  "weights": Array [
                    Object {
                      "countryId": 276,
                      "marketValue": 6000000,
                      "weight": 0.1,
                    },
                    Object {
                      "countryId": 246,
                      "marketValue": 12000000,
                      "weight": 0.2,
                    },
                    Object {
                      "countryId": 826,
                      "marketValue": 18000000,
                      "weight": 0.3,
                    },
                    Object {
                      "countryId": 643,
                      "marketValue": 24000000,
                      "weight": 0.4,
                    },
                  ],
                },
              ],
              "month": "2012-01",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Button
          asset="all"
          isActive="is-active"
          onClickHandler={[Function]}
/>,
        <Button
          asset="equity"
          isActive=""
          onClickHandler={[Function]}
/>,
        <Button
          asset="bond"
          isActive=""
          onClickHandler={[Function]}
/>,
        <Button
          asset="alternative"
          isActive=""
          onClickHandler={[Function]}
/>,
        <Button
          asset="property"
          isActive=""
          onClickHandler={[Function]}
/>,
        <Button
          asset="cash"
          isActive=""
          onClickHandler={[Function]}
/>,
        <Button
          asset="unknown"
          isActive=""
          onClickHandler={[Function]}
/>,
      ],
      "className": "tabs is-toggle is-centered is-fullwidth",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "asset": "all",
          "isActive": "is-active",
          "onClickHandler": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "asset": "equity",
          "isActive": "",
          "onClickHandler": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "asset": "bond",
          "isActive": "",
          "onClickHandler": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "class",
        "props": Object {
          "asset": "alternative",
          "isActive": "",
          "onClickHandler": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "class",
        "props": Object {
          "asset": "property",
          "isActive": "",
          "onClickHandler": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "class",
        "props": Object {
          "asset": "cash",
          "isActive": "",
          "onClickHandler": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "6",
        "nodeType": "class",
        "props": Object {
          "asset": "unknown",
          "isActive": "",
          "onClickHandler": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            asset="all"
            isActive="is-active"
            onClickHandler={[Function]}
/>,
          <Button
            asset="equity"
            isActive=""
            onClickHandler={[Function]}
/>,
          <Button
            asset="bond"
            isActive=""
            onClickHandler={[Function]}
/>,
          <Button
            asset="alternative"
            isActive=""
            onClickHandler={[Function]}
/>,
          <Button
            asset="property"
            isActive=""
            onClickHandler={[Function]}
/>,
          <Button
            asset="cash"
            isActive=""
            onClickHandler={[Function]}
/>,
          <Button
            asset="unknown"
            isActive=""
            onClickHandler={[Function]}
/>,
        ],
        "className": "tabs is-toggle is-centered is-fullwidth",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "asset": "all",
            "isActive": "is-active",
            "onClickHandler": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "asset": "equity",
            "isActive": "",
            "onClickHandler": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "asset": "bond",
            "isActive": "",
            "onClickHandler": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "class",
          "props": Object {
            "asset": "alternative",
            "isActive": "",
            "onClickHandler": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "class",
          "props": Object {
            "asset": "property",
            "isActive": "",
            "onClickHandler": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "class",
          "props": Object {
            "asset": "cash",
            "isActive": "",
            "onClickHandler": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "class",
          "props": Object {
            "asset": "unknown",
            "isActive": "",
            "onClickHandler": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
